core/View.php

```diff
@@ -336,9 +336,11 @@ public function setXFrameOptions($option = 'deny')
     {
         if ($option === 'deny' || $option === 'sameorigin') {
             $this->xFrameOptions = $option;
+            return;
         }
-        if ($option == 'allow') {
-            $this->xFrameOptions = null;
+        if (stripos($option, 'allow-from') !== false) {
```

Comment:
* you can edit the function docblock comment and add this new `allow-from` value, and an example of how to set it?